evalml/pipelines/components/transformers/imputers/time_series_imputer.py

```diff
@@ -155,35 +165,45 @@ def transform(self, X, y=None):
         if y is not None:
             y = infer_feature_types(y)
 
+        # This will change the logical type of BooleanNullable/IntegerNullable/AgeNullable columns with nans
+        # so we save the original schema to recreate it where possible after imputation
+        original_schema = X.ww.schema
+        X, y = self._handle_nullable_types(X, y)
+
         X_not_all_null = X.ww.drop(self._all_null_cols)
-        X_schema = X_not_all_null.ww.schema
-        X_schema = X_schema.get_subset_schema(
-            subset_cols=X_schema._filter_cols(
-                exclude=["IntegerNullable", "BooleanNullable", "AgeNullable"],
-            ),
```

Comment:
This change means that we'll no longer rely on woodwork inference for the nullable types after imputing. Instead, we're passing along these logical types even after imputation. 

This is something I've thought a lot on, since it doesn't immediately make sense that we wouldn't either continue to rely on type inference or specify the non nullable types that we know we can convert to after inference. 
- Don't rely on inference - this will take longer since type inference takes time, and it leaves the types up to woodwork type inference which, as we all know, can change at any time. Not leaving these types up to inference I think makes this component more efficient and more consistent.
- Keep the nullable types even after imputing - the goal of this epic is to be able to pass in nullable types wherever and whenever. We shouldn't need to avoid them anymore, so I don't want to add the extra complexity of determining what types to use unless it proves necessary. That might be if our dependencies show worse runtimes when continuing to use nullable types that are proving problematic for automl search times or if we ever choose to use the information we get from non nullable types that no nans can be present to optimize (ex - never try to impute non nullable data. Though, of course, that's at the imputer level where this conversation is happening, so it's a moot point).

Reply:
summarizing a convo from elsewhere: Performance testing to this point is showing the expected runtime improvements with no impact to model results.